xarray_jax/tests/xarray_jax_test.py

```diff
@@ -667,5 +667,63 @@ def test_assign_coords_arg_types(self):
         result.g.variable, xarray.Variable(('g2',), np.arange(2)))
 
 
+  def test_preserves_leaf_identity(self):
+    var = xarray.Variable(('x', 'y'), jnp.ones((2, 3)))
+    shape_dtype = jax.ShapeDtypeStruct(shape=(2, 3), dtype=jnp.float32)
+
+    test_leaves = [True, None, 42, 3.14, object(),
+                  jax.stages.ArgInfo(shape_dtype, lambda: "donated"), shape_dtype]
+
+    for leaf in test_leaves:
```

Comment:
Minor: this could be a good fit for a parameterized test, we typically use https://byoshimi2-abseil-py.readthedocs.io/en/latest/absl.testing.parameterized.html for this. What you have is similar in effect though so not essential.